### ISSUE TITLE
Font render failure.

### ISSUE THREAD

**Issue by ray-x** [hismailbulut/Neoray#19]
**Please complete the following information:**
 - OS or distro: MacOS
 - neoray version: Head

**Describe the bug**
Some font failed to render

**Steps to reproduce**
Can not render emoji
Also some font failed to render:

![image](https://user-images.githubusercontent.com/1681295/133290563-e9b398e3-4a37-41b5-b985-7d917d60c98a.png)
Left:  in kitty, same font
Right: neoray

**Your neoray config in init.vim** (Only the part that concerns neoray!)
```vim
vim.api.nvim_set_option('guifont', 'Victor Mono:h14')
```

**Crash log if available (may be generated by neoray after crash)**

**Debug output (you can generate starting neoray with --verbose option)**
[DEBUG] [NVIM] Neovim started with command: nvim --embed
[TRACE] [NVIM] Neovim version 0.6.0
[DEBUG] [NVIM] Attached to neovim as an ui client.
[TRACE] [NEORAY] Glfw version: 3.3.4 Cocoa NSGL EGL OSMesa
[DEBUG] [NEORAY] Monitor count: 1 Selected monitor: Built-in Retina Display
[DEBUG] [NEORAY] Video mode &{Width:1792 Height:1120 RedBits:8 GreenBits:8 BlueBits:8 RefreshRate:60}
[DEBUG] [NEORAY] Glfw window created with size 800 600
[DEBUG] [NEORAY] Monitor diagonal: 15.95 pdpi: 264.98 ldpi: 144.00
[DEBUG] [NEORAY] Using logical dpi.
[DEBUG] [NEORAY] Input callbacks are initialized.
[DEBUG] [RENDERER] Initializing opengl.
[TRACE] [RENDERER] Opengl Version: 4.1 ATI-3.10.22
[DEBUG] [RENDERER] Vendor: ATI Technologies Inc.
[DEBUG] [RENDERER] Renderer: AMD Radeon Pro 5300M OpenGL Engine
[DEBUG] [RENDERER] GLSL: 4.10
[DEBUG] [NEORAY] Loading default font.
[DEBUG] [NEORAY] Default font loaded.
[DEBUG] [NEORAY] Requesting deprecated options.
[DEBUG] [NEORAY] Checking user options.
[DEBUG] [NVIM] Option CursorAnimTime is 0.04
[DEBUG] [NVIM] Option Transparency is 0.95
[DEBUG] [NVIM] Option TargetTPS is 120
[DEBUG] [NVIM] Option ContextMenuOn is true
[DEBUG] [NVIM] Option WindowSize is 120 40
[DEBUG] [NEORAY] Loading font Victor Mono with size 14
[TRACE] [NEORAY] Regular: Victor Mono Regular Nerd Font Complete.ttf
[TRACE] [NEORAY] Bold Italic: Victor Mono Bold Italic Nerd Font Complete.ttf
[TRACE] [NEORAY] Italic: Victor Mono Italic Nerd Font Complete.ttf
[TRACE] [NEORAY] Bold: Victor Mono Bold Nerd Font Complete.ttf

**Comment by hismailbulut** [comment]
Currently neoray doesn't have an ability for rendering emojis, sorry for that. It seems like your font does not have this glyph and kitty uses some sort of another symbol font. Neoray currently can only render glyphs if it is in the specified user font or in default font.

**Comment by ray-x** [comment]
Yes. I am using nvim-web-devicons plugin. So some of the fonts are from devicons but not in nerd fonts.
Also noticed that the ligature is not working.

**Comment by reykjalin** [comment]
I have the same issue despite registering the Nerd Font:

```
[TRACE] [NEORAY] Regular: Fira Code Retina Nerd Font Complete.otf
[WARNING] [NEORAY] Font has no bold italic face.
[WARNING] [NEORAY] Font has no italic face.
[WARNING] [NEORAY] Font has no bold face.
```

<img width="903" alt="image" src="https://user-images.githubusercontent.com/13835680/135378814-bdd2d792-dde8-49c9-b72f-e3a38cb16ec2.png">

Here's the same part in my terminal using the same font:

<img width="903" alt="image" src="https://user-images.githubusercontent.com/13835680/135378870-818795f4-8781-4d6d-a141-6966a91c29b8.png">

Oddly enough, many other icons work in Neoray, so maybe it's just the arrow symbol that happens to be missing 🤔 

***

However, what matters more to me is that ligatures aren't supported:

<img width="499" alt="image" src="https://user-images.githubusercontent.com/13835680/135379084-09a923d6-d08d-44df-9c50-9d5f8c3f56fb.png">

Do you know if this is a limitation of the font rendering library that's being used?

**Comment by ray-x** [comment]
I solve this by using a manually patched nerdfont. But it took lots of efforts. Would be great neoray can load the graph vector from a built-in default font.

**Comment by hismailbulut** [comment]
This has just been fixed in the latest release. Neoray now supports font fallback.
